packages/runtimes/js/src/helpers/bundle-manifest.js

```diff
@@ -1,18 +1,20 @@
-var mapping = {};
+var mapping = new Map();
 
-function register(pairs) {
-  var keys = Object.keys(pairs);
-  for (var i = 0; i < keys.length; i++) {
-    mapping[keys[i]] = pairs[keys[i]];
+function register(baseUrl, manifest) {
+  for (var i = 0; i < manifest.length - 1; i += 2) {
+    mapping.set(manifest[i], {
+      baseUrl: baseUrl,
+      path: manifest[i + 1],
+    });
   }
 }
 
 function resolve(id) {
-  var resolved = mapping[id];
+  var resolved = mapping.get(id);
   if (resolved == null) {
     throw new Error('Could not resolve bundle with id ' + id);
   }
-  return resolved;
+  return new URL(resolved.path, resolved.baseUrl).toString();
```

Comment:
This one definitely won't work in IE. Probably ok at this point? I guess we can't simply concatenate the strings? `path` could probably start with `../` or something...

Reply:
Yeah not ideal. We could just recommend a polyfill for URL if anyone has issues I guess.